### PR TITLE
config: add --nofilelogging option.

### LINKED ISSUE #543: Allow disabling file-based logging
My use case: I have created a service on my operating system for dcrd.  This makes it easier to administer and allows it to be automatically started upon boot and even restarted after a crash.  It also automatically saves logs from the application's stdout, making the logging to $appdata/logs/ redundant.  I would like to turn it off, but there does not appear to be any way of currently doing so.

Opening this for dcrd but this same issue equally applies to dcrwallet, btcd, and btcwallet as well.

### PR DESCRIPTION
**Pull request by dnldd**
Fixes #543

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -603,21 +604,24 @@ func loadConfig() (*config, []string, error) {
...
+		cfg.LogDir = filepath.Join(cfg.LogDir, netName(activeNetParams))
+		// Initialize log rotation.  After log rotation has been initialized, the
+		// logger variables may be used.
+		initLogRotator(filepath.Join(cfg.LogDir, defaultLogFilename))
+	} else {
+		logRotator = nil
+	}
 
 	// Special show command to list supported subsystems and exit.
 	if cfg.DebugLevel == "show" {
```
Comment (chappjc) [inline config.go]: perhaps move this above `if !cfg.NoFileLogging {`

Comment (dnldd) [reply config.go]: sure, will do

File: config.go [hunk 2]
```diff
@@ -603,21 +604,24 @@ func loadConfig() (*config, []string, error) {
...
-	// Append the network type to the log directory so it is "namespaced"
-	// per network in the same fashion as the data directory.
-	cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
-	cfg.LogDir = filepath.Join(cfg.LogDir, netName(activeNetParams))
+	if !cfg.NoFileLogging {
+		// Append the network type to the log directory so it is "namespaced"
+		// per network in the same fashion as the data directory.
+		cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
+		cfg.LogDir = filepath.Join(cfg.LogDir, netName(activeNetParams))
+		// Initialize log rotation.  After log rotation has been initialized, the
```
Comment (davecgh) [inline config.go]: Need a blank line before this for consistency.

Comment (dnldd) [reply config.go]: updated.

**Pull request merged**